### PR TITLE
Add build manifest

### PR DESCRIPTION
**Pull request by chcosta**
- Creates a build manifest during the build which later stages / jobs (or different pipelines) can consume.

- Modifies signing validation to be an MSBuild task which can consume the manifest and is more extensible than the console tool previously used

- Splits out many signing property defaults from the Arcade Sdk so they can be imported as needed

- Replaces outdated Microsoft.Bcl.JsonSources with System.Text.Json

`_UseBuildManifest` in `eng\common\variables.yml` is currently set to `False` because we need to produce a version of the SDK with these changes before we can enable the build manifest.

Validated with https://dnceng.visualstudio.com/internal/_build/results?buildId=590601&view=results

### CODE REVIEW

File: eng/common/build.ps1 [hunk 1]
```diff
@@ -61,6 +62,7 @@ function Print-Usage() {
   Write-Host "  -prepareMachine         Prepare machine for CI run, clean up processes after build"
   Write-Host "  -warnAsError <value>    Sets warnaserror msbuild parameter ('true' or 'false')"
   Write-Host "  -msbuildEngine <value>  Msbuild engine to use to run build ('dotnet', 'vs', or unspecified)."
+  Write-Host "  -manifest               Generates a build manifest, including signing metadata.  Requires '-sign' and 'publish' also be specified."
```
Comment (tmat) [inline eng/common/build.ps1:65]: > Write-Host "  -manifest               Generates a build manifest, including signing metadata.  Requires '-sign' and 'publish' also be specified." [](start = 1, length = 146)

Do we need a switch? Why not just do it for all CI runs?

Comment (tmat) [reply eng/common/build.ps1:65]: I'd very much prefer that since adding another switch means PR validation diverging from official builds. Better keep then as close as possible.

---
In reply to: [404448797](https://github.com/dotnet/arcade/pull/5210#discussion_r404448797) [](ancestors = 404448797)

File: eng/common/sdk-task.ps1 [hunk 2]
```diff
@@ -57,6 +57,18 @@ try {
...
+  if( $msbuildEngine -eq "vs") {
+    # Ensure desktop MSBuild is available for sdk tasks.
+    if( -not ($GlobalJson.tools.PSObject.Properties.Name -match "vs" )) {
+      $GlobalJson.tools | Add-Member -Name "vs" -Value (ConvertFrom-Json "{ `"version`": `"16.4`" }") -MemberType NoteProperty
+    }
+    if( -not ($GlobalJson.tools.PSObject.Properties.Name -match "xcopy-msbuild" )) {
+      $GlobalJson.tools | Add-Member -Name "xcopy-msbuild" -Value "16.4.0-alpha" -MemberType NoteProperty
+    }
+
+    InitializeVisualStudioMSBuild $true
```
Comment (tmat) [inline eng/common/sdk-task.ps1:69]: >InitializeVisualStudioMSBuild [](start = 4, length = 29)

The function takes $vsRequirements optional param where you can specify the version.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj [hunk 3]
```diff
@@ -256,7 +258,7 @@
              Condition="'@(_SolutionBuildTargets)' != ''" />
 
     <MSBuild Projects="AfterSolutionBuild.proj"
-             Properties="@(_CommonProps);_NETCORE_ENGINEERING_TELEMETRY=Build"
+             Properties="@(_CommonProps);_NETCORE_ENGINEERING_TELEMETRY=Build;"
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj:261]: >; [](start = 77, length = 1)

Unnecessary?

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj [hunk 4]
```diff
@@ -30,6 +30,7 @@
     Pack                            "true" to build NuGet packages and VS insertion manifests
     Sign                            "true" to sign built binaries
     Publish                         "true" to publish artifacts (e.g. symbols)
+    GenerateBuildManifest           "true" to generate a build manifest during publish step.  Requires "Sign" and "Publish" are also true so that signing metadata can be included in the build manifest.
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj:33]: >GenerateBuildManifest [](start = 4, length = 21)

I think we should generate the manifest when ContinuousIntegrationBuild == true

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Manifest.targets [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
+<Project>
+  <PropertyGroup>
+    <AzureDevOpsBuildTasksAssembly Condition="'$(MSBuildRuntimeType)' != 'Core'">$(NuGetPackageRoot)Microsoft.DotNet.AzureDevOps.Build\$(ArcadeSdkVersion)\tools\net472\Microsoft.DotNet.AzureDevOps.Build.dll</AzureDevOpsBuildTasksAssembly>
+    <AzureDevOpsBuildTasksAssembly Condition="'$(MSBuildRuntimeType)' == 'Core'">$(NuGetPackageRoot)Microsoft.DotNet.AzureDevOps.Build\$(ArcadeSdkVersion)\tools\netcoreapp2.1\Microsoft.DotNet.AzureDevOps.Build.dll</AzureDevOpsBuildTasksAssembly>
+  </PropertyGroup>
+
+  <Import Project="$(MSBuildThisFileDirectory)Sign.props" />
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Manifest.targets:8]: >$(MSBuildThisFileDirectory) [](start = 19, length = 27)

No need to qualify.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj [hunk 6]
```diff
@@ -237,5 +238,5 @@
                     PdbConversionTreatAsWarning=""
                     Condition="$(PublishToSymbolServer)"/>
   </Target>
-
+  <Import Condition="$(GenerateBuildManifest)" Project="$(MSBuildThisFileDirectory)Manifest.targets" />
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj:241]: >MSBuildThisFileDirectory [](start = 58, length = 24)

`MSBuildThisFileDirectory` not necessary

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/GenerateBuildManifest.cs [hunk 7]
```diff
@@ -1,89 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
```
Comment (chcosta) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/GenerateBuildManifest.cs:1]: Unintentional delete, I'll add this back

**Comment by chcosta** [comment]
Latest changes validated internally with https://dev.azure.com/dnceng/internal/_build/results?buildId=592451&view=logs&j=0a6c679f-72be-5867-422e-acb741a068a3&t=d8e44ddf-8ea2-5518-25e7-4300471a271a

File: .gitignore [hunk 8]
```diff
@@ -6,6 +6,7 @@ syntax: glob
 /[Tt]ools/
```
Comment (tmat) [inline .gitignore:6]: Remove?

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj [hunk 9]
```diff
@@ -237,5 +237,5 @@
                     PdbConversionTreatAsWarning=""
                     Condition="$(PublishToSymbolServer)"/>
   </Target>
-
+  <Import Project="Manifest.targets" />
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj:240]: >Manifest [](start = 19, length = 8)

Since Manifest.target only has a single simple target I'd just put it here directly.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Sign.proj [hunk 10]
```diff
@@ -4,52 +4,7 @@
...
-    <AllowEmptySignList>false</AllowEmptySignList>
-  </PropertyGroup>
-
-  <PropertyGroup>
-    <NETCORE_ENGINEERING_TELEMETRY>Signing</NETCORE_ENGINEERING_TELEMETRY>
-  </PropertyGroup>
-
-  <!-- Allow repository to customize signing configuration -->
-  <Import Project="$(RepositoryEngineeringDir)Signing.props" Condition="Exists('$(RepositoryEngineeringDir)Signing.props')" />
+  <Import Project="$(MSBuildThisFileDirectory)Sign.props" />
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Sign.proj:7]: >MSBuildThisFileDirectory [](start = 21, length = 24)

No need to qualify.

**Comment by tmat** [comment]
>       AssetManifestPath="$(AssetManifestFilePath)" />

Is this going away in future? Seems like we have now two manifests. Do we need both?

---
Refers to: src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj:145 in 67498d5. [](commit_id = 67498d50862d0489c6e578b888535b767fee6326, deletion_comment = False)

**Comment by chcosta** [comment]
They're serving two slightly different purposes.

Build manifest - defines the artifacts produced by the build and metadata for how the build believes they should be used (signing info at the moment).

Asset manifest - defines the artifacts that were published externally and where they can be found.

There's some overlap here in the sense that the `PushToAzureDevOpsArtifacts` task could consume the build manifest which it would then use to produce the asset manifest.  

I agree, it would be nice if there was one manifest (or they were named better), but I don't yet have an idea for how we could combine these.

@JohnTortugo @mmitche

**Comment by chcosta** [comment]
Perhaps, if the build manifest contained the metadata and rather than using the asset manifest as the "here's what I actually did" document, we used the build manifest as the "here's what you should do" document and then later we trusted that it actually happened, then we could eliminate the asset manifest.

**Comment by mmitche** [comment]
The asset manifest actually isn't quite that unfortunately. It doesn't define entirely what was done. It's a combo of "what should I do" and "what I did". The asset manifest is produced prior to publishing and contains a list of artifacts. But it also contains information for publishing that identifies where those artifacts should go if the build happens to be published. The big one there is for blobs. The azdo build artifacts have the files that got produced, but the asset manifest tells us what subpaths those artifacts should go live on if they happen to be finally published. Publishing defines the base endpoints (e.g. dotnetcli.blob.core.windows.net/dotnet) and then the relative paths are defined by the asset manifests

@JohnTortugo and I were talking about the possibility of getting rid of the asset manifest the other day. Much of what the asset manifest contains is duplicated in the BAR build info. But the asset manifest also contains info that is not duplicated, and potentially could change over time as we change publishing. In order to get rid of the manifest, you need to encapsulate everything in there today in BAR, and make it extensible enough such that changes to that info do not involve a database change every time. The xml based manifest is pretty perfect for that.

>There's some overlap here in the sense that the PushToAzureDevOpsArtifacts task could consume the build manifest which it would then use to produce the asset manifest.

This actually could work really well. What if we called the build manifest in this case a "signing manifest" (since that's what it's primarily targeted at) and then that info gets added to the asset manifest like you say. We also add to the asset manifest, a version number.  Let's call it version 2, where version one of the asset manifest is the current 3.x and 5.x builds.

This version of the asset manifest can be interpreted **differently** by the publishing infrastructure. For the most part, everything is identical. But in order to sign after the build, we'll need the signing info available, so we also need to archive the manifest itself (since builds will go away in AzDO after N days),

To recap:

So version 1 of publishing, the current state, knows nothing about adding signing info to the asset manifest, and version 2, which the new PushToAzureDevOps task will generate, will include the signing info from the provided signing manifest. Publishing will be altered to know about both asset versions. For version 1, it does what it does today.  For version 2, it knows that the artifacts are not signed, will need to be signed later, and thus archives the manifest in blob storage or wherever for use by the post-build prep stages where signing is being moved to.

File: eng/common-variables.yml [hunk 11]
```diff
@@ -18,6 +20,8 @@ variables:
       value: True
     - name: _SignType
       value: real
+    - name: _PublishBuildManifest
```
Comment (JohnTortugo) [inline eng/common-variables.yml:23]: Isn't `False` the default -- line 14?

File: eng/common/templates/job/job.yml [hunk 12]
```diff
@@ -218,3 +219,12 @@ jobs:
         ArtifactName: AssetManifests
       continueOnError: ${{ parameters.continueOnError }}
       condition: and(succeeded(), eq(variables['_DotNetPublishToBlobFeed'], 'true'))
+
+  - ${{ if eq(parameters.useBuildManifest, true) }}:
+    - task: PublishBuildArtifacts@1
+      displayName: Publish Build Manifest
+      inputs:
+        PathToPublish: '$(Build.SourcesDirectory)/artifacts/log/$(_BuildConfig)/manifest.props'
```
Comment (JohnTortugo) [inline eng/common/templates/job/job.yml:227]: IMHO we really need a better name than just "manifest.props"

File: eng/common/templates/job/job.yml [hunk 13]
```diff
@@ -24,6 +24,7 @@ parameters:
   enablePublishBuildAssets: false
   enablePublishTestResults: false
   enablePublishUsingPipelines: false
+  useBuildManifest: false
```
Comment (JohnTortugo) [inline eng/common/templates/job/job.yml:27]: Looks like this parameter actually means "Publish Build Manifest"

File: eng/common/templates/post-build/post-build.yml [hunk 14]
```diff
@@ -113,6 +114,16 @@ stages:
     pool:
       vmImage: 'windows-2019'
     steps:
+      - ${{ if eq(parameters.useBuildManifest, true) }}:
```
Comment (JohnTortugo) [inline eng/common/templates/post-build/post-build.yml:117]: ~~Perhaps I'm blind but where is `BuildManifests\` being used in this job?~~

File: src/Microsoft.DotNet.Arcade.Sdk/Microsoft.DotNet.Arcade.Sdk.csproj [hunk 15]
```diff
@@ -12,7 +12,7 @@
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
 
     <Description>Common toolset for repositories</Description>
-    <PackageTags>Roslyn Build Repository Toolset MSBuild SDK</PackageTags>
+    <PackageTags>Arcade Build Repository Toolset MSBuild SDK</PackageTags>
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Arcade.Sdk/Microsoft.DotNet.Arcade.Sdk.csproj:15]: Wow, it took us just 2 years to do that haha

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Manifest.targets [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+    <AzureDevOpsBuildTasksAssembly Condition="'$(MSBuildRuntimeType)' != 'Core'">$(NuGetPackageRoot)Microsoft.DotNet.AzureDevOps.Build\$(ArcadeSdkVersion)\tools\net472\Microsoft.DotNet.AzureDevOps.Build.dll</AzureDevOpsBuildTasksAssembly>
+    <AzureDevOpsBuildTasksAssembly Condition="'$(MSBuildRuntimeType)' == 'Core'">$(NuGetPackageRoot)Microsoft.DotNet.AzureDevOps.Build\$(ArcadeSdkVersion)\tools\netcoreapp2.1\Microsoft.DotNet.AzureDevOps.Build.dll</AzureDevOpsBuildTasksAssembly>
+  </PropertyGroup>
+
+  <Import Project="Sign.props" />
+  <UsingTask TaskName="Microsoft.DotNet.AzureDevOps.Build.GenerateAzureDevOpsBuildManifest" AssemblyFile="$(AzureDevOpsBuildTasksAssembly)"/>
+
+  <Target Name="GenerateBuildManifest">
+    <GenerateAzureDevOpsBuildManifest
+      AzureDevOpsCollectionUri="$(SYSTEM_TEAMFOUNDATIONCOLLECTIONURI)"
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Manifest.targets:13]: NIT: Can you retrieve these AzDO variables from the environment inside the task? That way you could set a default value so you don't have to set them here. IMO it makes the code cleaner / easy to read.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj (hunk 9 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj:240]: Assuming we'll keep using two manifests for now.. wouldn't it make more sense to put the generation of this "signing" manifest in the Sign.proj file?

File: src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SigningValidation.proj [hunk 17]
```diff
@@ -14,45 +14,44 @@
...
+      Recursive="true"
+      FileStatus="AllFiles"
+      InputFiles="@(ItemsToSign)"
+      ExclusionsFile="$(SignCheckExclusionsFile)"
+      EnableJarSignatureVerification="$(EnableJarSigningCheck)"
+      VerifyStrongName="$(EnableStrongNameCheck)"
+      LogFile="$(SignCheckLog)"
+      ErrorLogFile="$(SignCheckErrorLog)"
+      ArtifactFolder="$(SignCheckInputDir)"
+      Condition="'$(BuildManifestFile)' != ''"/>
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SigningValidation.proj:54]: Perhaps I didn't get the big picture yet: Are we going to disable signing validation for everyone that doesn't set `_PublishBuildManifest` to `true`?

File: src/Microsoft.DotNet.AzureDevOps.Build/src/GenerateBuildManifest.cs [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+{
+    public class GenerateAzureDevOpsBuildManifest : Microsoft.Build.Utilities.Task
+    {
+        [Required]
+        public string AzureDevOpsCollectionUri { get; set; }
+        [Required]
+        public string AzureDevOpsProject { get; set; }
+        [Required]
+        public int AzureDevOpsBuildId { get; set; }
+        public string ManifestPath { get; set; }
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.AzureDevOps.Build/src/GenerateBuildManifest.cs:19]: Is this one not required?

File: src/Microsoft.DotNet.AzureDevOps.Build/src/GenerateBuildManifest.cs [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+            var itemGroup = project.CreateItemGroupElement();
+            project.AppendChild(itemGroup);
+            if (ItemsToSign != null)
+            {
+                foreach (var itemToSign in ItemsToSign)
+                {
+                    var filename = itemToSign.ItemSpec.Replace('\\', '/');
+                    {
+                        var metadata = itemToSign.CloneCustomMetadata() as Dictionary<string, string>;
+                        var fileExtensionSignInfo = FileExtensionSignInfo.FirstOrDefault(f => Path.GetExtension(itemToSign.ItemSpec).Equals(f.ItemSpec, StringComparison.InvariantCultureIgnoreCase));
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.AzureDevOps.Build/src/GenerateBuildManifest.cs:41]: Perhaps I'm blind here but where are the locals `fileExtensionSignInfo` and `fileSignInfo` used?

File: src/Microsoft.DotNet.AzureDevOps.Build/src/GenerateBuildManifest.cs [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.AzureDevOps.Build
+{
+    public class GenerateAzureDevOpsBuildManifest : Microsoft.Build.Utilities.Task
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.AzureDevOps.Build/src/GenerateBuildManifest.cs:11]: Wouldn't make sense to put this task in the SignTool project?

File: eng/common/templates/post-build/post-build.yml [hunk 21]
```diff
@@ -139,7 +150,7 @@ stages:
         displayName: Validate
         inputs:
           filePath: eng\common\sdk-task.ps1
-          arguments: -task SigningValidation -restore -msbuildEngine dotnet
+          arguments: -task SigningValidation -restore -msbuildEngine vs
```
Comment (JohnTortugo) [inline eng/common/templates/post-build/post-build.yml:153]: I think we need to make clear here that SigningValidation uses the `BuildManifests` artifact downloaded above.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj (hunk 9 above)
Comment (chcosta) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj:240]: Moving this into Publish.proj sounds good to me.  I had debated making this change.

File: eng/common-variables.yml (hunk 11 above)
Comment (chcosta) [reply eng/common-variables.yml:23]: I made a comment about this in the description 

> _UseBuildManifest in eng\common\variables.yml is currently set to False because we need to produce a version of the SDK with these changes before we can enable the build manifest.

File: eng/common/templates/job/job.yml (hunk 12 above)
Comment (chcosta) [reply eng/common/templates/job/job.yml:227]: what do you think it should be?

File: src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SigningValidation.proj (hunk 17 above)
Comment (chcosta) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SigningValidation.proj:54]: Good catch!  This was from a previous design that I didn't clean up

File: src/Microsoft.DotNet.AzureDevOps.Build/src/GenerateBuildManifest.cs (hunk 20 above)
Comment (chcosta) [reply src/Microsoft.DotNet.AzureDevOps.Build/src/GenerateBuildManifest.cs:11]: That would limit the use of the build manifest in the future to just signing.  You couldn't, for example, create a build manifest if you didn't sign but wanted to publish.  In order to not design myself into a corner where the build manifest really just meant "signing manifest", I opted to create a new package.  I internally debated this one for quite a while (SignTool project, Feed project, Arcade SDK, etc...).

File: eng/common-variables.yml (hunk 11 above)
Comment (JohnTortugo) [reply eng/common-variables.yml:23]: My bad then!

File: eng/common/templates/job/job.yml (hunk 13 above)
Comment (chcosta) [reply eng/common/templates/job/job.yml:27]: I couldn't decide on this either.  It means both "publish" and "consume".  I'll switch it back to "_UseBuildManifest".

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Manifest.targets (hunk 16 above)
Comment (chcosta) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/Manifest.targets:13]: Baking in environment variables to the task makes updates here more difficult and creates a black box of magic.  I prefer to be explicit about environment variable usage (I know it's not always possible).

File: eng/common/templates/post-build/post-build.yml (hunk 21 above)
Comment (chcosta) [reply eng/common/templates/post-build/post-build.yml:153]: SigningValidation will also work if no build manifest is provided (old way of doing things).  I'll add a note.

**Comment by chcosta** [comment]
Latest changes validated with https://dev.azure.com/dnceng/internal/_build/results?buildId=593679&view=results

File: eng/common/templates/job/job.yml (hunk 12 above)
Comment (JohnTortugo) [reply eng/common/templates/job/job.yml:227]: I'll post a more general observation at the PR.

File: eng/common/templates/post-build/post-build.yml (hunk 21 above)
Comment (JohnTortugo) [reply eng/common/templates/post-build/post-build.yml:155]: What I didn't like here was that the SigningValidation task was referring to `$(BUILD_ARTIFACTSTAGINGDIRECTORY)/BuildManifests` and it wasn't apparent in this call. Also, it diverge of the way the current parameters are informed, e.g., PackageBasePath.

File: src/Microsoft.DotNet.AzureDevOps.Build/src/GenerateBuildManifest.cs (hunk 20 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.AzureDevOps.Build/src/GenerateBuildManifest.cs:11]: I'll add a more general comment in the PR level.

File: src/SignCheck/SignCheck/SignCheckTask.cs [hunk 22]
```diff
@@ -0,0 +1,145 @@
...
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace SignCheck
+{
+    public class SignCheckTask : Microsoft.Build.Utilities.Task
+    {
+        public bool EnableJarSignatureVerification
```
Comment (JohnTortugo) [inline src/SignCheck/SignCheck/SignCheckTask.cs:12]: I believe the style we use overall is to keep the get & set on the same line.

File: src/SignCheck/SignCheck/SignCheckTask.cs [hunk 23]
```diff
@@ -0,0 +1,145 @@
+using Microsoft.Build.Framework;
```
Comment (JohnTortugo) [inline src/SignCheck/SignCheck/SignCheckTask.cs:1]: Missing the copyright header.

File: src/SignCheck/SignCheck/SignCheckTask.cs [hunk 24]
```diff
@@ -0,0 +1,145 @@
...
+                                Log.LogError($" - {file}");
+                            }
+                        }
+                    }
+                }
+            }
+            options.InputFiles = inputFiles.ToArray();
+
+            LogVerbosity verbosity;
+            if(Enum.TryParse<LogVerbosity>(Verbosity, out verbosity))
```
Comment (JohnTortugo) [inline src/SignCheck/SignCheck/SignCheckTask.cs:134]: NIT: can inline the verbosity var declaration

File: src/SignCheck/SignCheck/SignCheckTask.cs [hunk 25]
```diff
@@ -0,0 +1,145 @@
...
+            {
+                options.Verbosity = verbosity;
+            }
+            
+            var sc = new SignCheck(new string[] { });
+            sc.Options = options;
+            int result = sc.Run();
+            return result != 0 || Log.HasLoggedErrors ? false: true;
```
Comment (JohnTortugo) [inline src/SignCheck/SignCheck/SignCheckTask.cs:142]: NIT: `return (result == 0 && !Log.HasLoggedErrors);`

File: src/SignCheck/SignCheck/SignCheckTask.cs [hunk 26]
```diff
@@ -0,0 +1,145 @@
...
+            {
+                foreach (var checkFile in InputFiles.Select(s => s.ItemSpec).ToArray())
+                {
+                    if (Path.IsPathRooted(checkFile))
+                    {
+                        inputFiles.Add(checkFile);
+                    }
+                    else
+                    {
+                        ArtifactFolder = ArtifactFolder ?? Environment.CurrentDirectory;
```
Comment (JohnTortugo) [inline src/SignCheck/SignCheck/SignCheckTask.cs:108]: NIT: can move these two lines out of the for loop.

**Comment by mmitche** [comment]
@chcosta I'd like to get this merged into the asset manifest during the publish to BAR stage as outlined above (with a BAR manifest verison), rather than carrying along a separate manifest. What would it take to get that done?

**Comment by chcosta** [comment]
Shouldn't be too difficult. I'm hoping to have this merged today but I think I need to make one more change to this pr before merging

**Comment by JohnTortugo** [comment]
I fully agree with what @mmitche said. IMHO having two manifests isn't ideal. 

> I agree, it would be nice if there was one manifest (or they were named better), but I don't yet have an idea for how we could combine these.

I think we need to have a plan for this. What other information will this manifest contains in the future? Right now I see that it only contains signing information.

**Comment by chcosta** [comment]
I think we all agree that two manifests isn't ideal.

**Comment by mmitche** [comment]
@chcosta Did you open an issue to merge the manifests?

**Comment by chcosta** [comment]
I asked @jonfortescue to open one, as he's working on that right now

**Comment by jonfortescue** [comment]
@chcosta I am indeed working on it and I also forgot to open the issue to show that so i'll go do that now!
